json_structs.go

```diff
@@ -83,16 +84,29 @@ type albumImage struct {
 		} `json:"LargestVideo"`
 	} `json:"Uris"`
 
-	fileDatetime  time.Time
-	builtFilename string // The final filename, after template replacements
+	fileDatetime                  time.Time
+	builtFilename                 string // The final filename, after template replacements
+	PreferredDisplayFileExtension string `json:"PreferredDisplayFileExtension"`
 }
 
 func (a *albumImage) buildFilename(tmpl *template.Template) error {
 	replacementVars := map[string]string{
-		"FileName":    a.FileName,
-		"ImageKey":    a.ImageKey,
-		"ArchivedMD5": a.ArchivedMD5,
-		"UploadKey":   a.UploadKey,
+		"FileName":      a.FileName,
+		"ImageKey":      a.ImageKey,
+		"ArchivedMD5":   a.ArchivedMD5,
+		"UploadKey":     a.UploadKey,
+		"FileNameNoExt": a.FileName,
+		"Extension":     a.PreferredDisplayFileExtension,
```

Comment:
that's ok, though I'd prefer to calculate values then create `replacementVars`, I think it's more clear to understand instead of assigning a value, then eventually change it later